spec/controllers/adopters_controller_spec.rb

```diff
@@ -318,13 +308,215 @@
+        end
+      end
+    end
+  end
+
+  describe 'GET #search' do
+    context 'se listan los adoptantes buscados' do
+
+      let!(:adToSearch) { create(:adopter, first_name: 'Miguel', last_name: 'Viera') }
+      let!(:adToSearch2) { create(:adopter, first_name: 'Charly', last_name: 'Sosa', blacklisted: 'true') }
+      let!(:adToSearch3) { create(:adopter, first_name: 'Char', last_name: 'Mate', blacklisted: 'true') }
+      let!(:adToSearch4) { create(:adopter, first_name: 'Charly', last_name: 'Sosa', blacklisted: 'false') }
+
+      before(:each) do
+        request.headers['X-USER-TOKEN'] = user.authentication_token
+      end
+
+      context 'por nombre especifico' do
+        before(:each) do
+          get :search, name: 'Miguel', format: 'json'
+        end
+
+        it 'devuelve codigo 200' do
+          expect(response.status).to eq(200)
+        end
+
+        it 'devuelve un array no vacio' do
+          expect(parse_response(response)['adopters']).to_not be_nil
+        end
+
+        it 'devuelve el adoptante buscado' do
+          expect(parse_response(response)['adopters'][0]['id']).to eq(adToSearch.id)
+        end
+      end
+
+      context 'por nombre parcial en lista negra' do
+        before(:each) do
+          get :search, name: 'Cha', blacklisted: 'true', format: 'json'
+        end
+
+        it 'devuelve codigo 200' do
+          expect(response.status).to eq(200)
+        end
+
+        it 'devuelve un array no vacio' do
+          expect(parse_response(response)['adopters']).to_not be_nil
+        end
+
+        it 'devuelve los adoptantes encontrados' do
+          @id1 = parse_response(response)['adopters'][0]['id']
+          @id2 = parse_response(response)['adopters'][1]['id']
+          @arrids = [@id1,@id2]
+          expect(@arrids).to match_array([adToSearch2.id,adToSearch3.id])
+        end
+      end
+
+      context 'por nombre especifico en lista negra' do
+        before(:each) do
+          get :search, name: 'Charly', blacklisted: 'true', format: 'json'
+        end
+
+        it 'devuelve codigo 200' do
+          expect(response.status).to eq(200)
+        end
+
+        it 'devuelve un array no vacio' do
+          expect(parse_response(response)['adopters']).to_not be_nil
+        end
+
+        it 'devuelve el adoptante de la lista negra' do
+          expect(parse_response(response)['adopters'][0]['id']).to eq(adToSearch2.id)
+        end
+      end
+    end
+  end
+
+  describe 'PUT #set_as_blacklisted' do
+
+    let!(:adopter3) { create(:adopter) }
+
+    context 'exitoso' do
+      before(:each) do
+        request.headers['X-USER-TOKEN'] = user.authentication_token
+        put :set_as_blacklisted, id: adopter3.id, format: 'json'
+      end
+
+      it 'devuelve codigo 204 No Content' do
+        expect(response.status).to eq(204)
+      end
+
+      it 'no se envia al adoptante a la blacklist' do
+        expect(adopter3.reload.blacklisted).to_not eq('false')
```

Comment:
Acá justamente, el adopter3 fue seteado como blacklisted, o sea, que está en la lista negra.
Campaz que esta línea se puede poner así y queda mas fácil de interpretar:
`expect(adopter3.reload.blacklisted).to eq('true')`